cdk/api.py

```diff
@@ -58,25 +59,22 @@ def __init__(self, scope: core.Construct, id: str,
             ]
         )
 
-        on_demand_function.add_environment(
+        on_demand_function.function.add_environment(
             "WEBHOOK_ENDPOINT_URL",
-            (f"https://{self.api.rest_api_id}.execute-api.{self.region}"
-             f".amazonaws.com/{self.lambda_release_alias}/new_recording")
+            (f"https://{self.api.rest_api_id}.execute-api.{core.Stack.of(self).region}"
+             f".amazonaws.com/{lambda_release_alias}/new_recording")
         )
 
     def add_monitoring(self, monitoring):
 
-        for metric_name in ["4XXError", "5XXError"]:
-            for resource in [self.new_recording_resource, self.ingest_resource]:
-                alarm = cloudwatch.Alarm(self, f"{metric_name}Alarm",
+        for metric_name in ["4XX", "5XX"]:
+                alarm = cloudwatch.Alarm(self, f"ZoomWebhook{metric_name}Alarm",
                     metric=cloudwatch.Metric(
                         metric_name=metric_name,
                         namespace="AWS/ApiGateway",
                         dimensions={
                             "ApiName": self.rest_api_name,
-                            "Stage": "live",
-                            "Method": "POST",
-                            "Resource": resource.path,
```

Comment:
you can just have one 4xx and 5xx alarm for the whole api... if you want to have them per resource... which actually is probably a good idea so we can stop getting annoying emails everytime somebody puts the wrong link into the Zoom+ tool then the metric names have to be unique

Reply:
i decided to do them separately since we could. fixed the resource ids.